lib/zold/score.rb

```diff
@@ -157,22 +161,18 @@ def reduced(max = 4)
 
     def next
       raise 'This score is not valid' unless valid?
-      idx = 0
-      loop do
-        suffix = idx.to_s(16)
-        score = Score.new(
-          time: time, host: host, port: port, invoice: invoice, suffixes: suffixes + [suffix],
-          strength: strength
-        )
-        return score if score.valid?
-        if score.expired?
-          return Score.new(
-            time: Time.now, host: host, port: port, invoice: invoice,
-            suffixes: [], strength: strength
-          )
-        end
-        idx += 1
-      end
+      return Score.new(
+        time: Time.now, host: host, port: port, invoice: invoice,
+        suffixes: [], strength: strength
+      ) if expired?
+      idx = ScoreExt.calculate_nonce_multi_core(
```

Comment:
I would remove the _multi_core part from the method name, it is not adding anything (maybe do we have a calculate_nonce_single_code method?) and makes the corder harder to read

Reply:
Logic goes over style at this point.

Reply:
It is not a matter of style, but to properly name a method. Also, you will have to change the signature of the method anyway while you remove the _core_ param, I think you can easily change the name of the method.

Reply:
Sorry, just have put the comment in the wrong place. Renaming makes sense here.